python/tvm/relax/frontend/nn/op.py

```diff
@@ -1461,13 +1461,87 @@ def _convert(arg):
 OutType = TypeVar("OutType", bound=Union[Tensor, Sequence[Tensor]])
 
 
+def tensor_ir_op(
+    func: _tir.PrimFunc,
+    name_hint: str,
```

Comment:
There’s a bit of complication here: if the PrimFunc provided is a public function (has “global_symbol” field in its attrs), Relax is not allowed to rename it, and in this case, it’s not a name hint but a name instead. Therefore, we will have to check symbol duplication and potentially throw an error if it happens.

Reply:
We could probably leave this logic to future work, but let’s rename name_hint to name to better reflect this point

Reply:
I agree and thanks for pointing it out. However, the current Python interface `AddFunction` also treats it as `name_hint`, which may be renamed if conflicts exist.

It would be an independent problem out of the scope of this PR.